recipe/meta.yaml

```diff
@@ -0,0 +1,73 @@
+{% set name = "pypdfium2" %}
+{% set version = "4.18.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/pypdfium2-team/{{ name }}/releases/download/{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: c937121dc475942697fbb3e04ffa7b28d36afc2b76cc9aac22fbd327c6dc6d61
+  patches:
+    - 0001-use-pdfium-library-from-repack-dependency.patch
+    - 0002-copy-binding-file-from-ctypesgen-output-path.patch
+    - 0003-fix-windows-include-location.patch
+
+build:
+  skip: True  # [linux and (ppc64le or s390x)]
+  entry_points:
+    - pypdfium2 = pypdfium2.__main__:cli_main
+  script:
+    - {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  number: 0
+
+requirements:
+  build:
+    - git         # [not win]
+    - {{ compiler('c') }}
```

Comment:
In the build logs, I see
```
INFO: Status: Preprocessing /tmp/tmprlu8i60_.h
INFO: Status: gcc -E -U __GNUC__ -dD "-D__extension__=" "-D__const=const" "-D__asm__(x)=" "-D__asm(x)=" "-DCTYPESGEN=1" "/tmp/tmprlu8i60_.h"
ERROR: gcc -E: /bin/sh: gcc: command not found
```

Reply:
This might currently be masked by the patch to use the reference bindings?
I din't check if that code is truly active, but if it is, then the ctypesgen call's result would be unused.

Reply:
what we should do exactly ?

Reply:
> Import test is passing, so the build is doing something which is not needed, but that would not affect end user

Yes, this would be my understanding.

> what we should do exactly ?

Certainly not what you are currently doing. The whole patching is thoroughly messed up, unsustainable and should be revised from the ground, as stated before: https://github.com/AnacondaRecipes/pypdfium2-feedstock/pull/2#issuecomment-1698031058, https://github.com/bblanchon/pdfium-binaries/issues/119#issuecomment-1697474571

I would recommend to just use `$PDFIUM_PLATFORM=none` and deploy the bindings externally in the recipe's script section so you don't need any patches in the first place.

However, if you really do want to run custom code in our setup infra, you should start at the top-level `setup.py` by adding a new case in `packaging_handler()`, e.g. `elif target == "conda-thirdparty"` and do your work there (deploying bindings & binary into `src/pypdfium2/` and calling `setuptools.setup(...)` at the end).
Then you would define `PDFIUM_PLATFORM="conda-thirdparty"` in the recipe env to use this new code passage.
If done correctly, the patch should effectively be additions-only and really short.

Apart from that, I'm wondering where you currently get the binary from? I suppose it's ctypesgen's library loader finding it automatically from the env's sysroot, where supposedly the pdfium-binaries are installed to?
This should ideally be made explicit by symlinking or passing ctypesgen the right runtime libdirs.

Reply:
Well, the patch will be a burden to maintain, for one thing.
If this is a "rebuild once in a blue moon" project, then maybe it's doable, but technically it's just a terrible mess, while it should be comparatively easy to get right.

Reply:
Ah, I just noticed @skupr-anaconda changed it to use the approach I suggested above. That looks substantially better, thanks!